crd/multitenantpodnetworkconfig/api/v1alpha1/multitenantpodnetworkconfig.go

```diff
@@ -0,0 +1,62 @@
+//go:build !ignore_uncovered
+// +build !ignore_uncovered
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// +kubebuilder:object:root=true
+
+// MultitenantPodNetworkConfig is the Schema for the multitenantpodnetworkconfigs API
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:resource:shortName=mpnc
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels=managed=
+// +kubebuilder:metadata:labels=owner=
+// +kubebuilder:printcolumn:name="PodNetworkInstance",type=string,JSONPath=`.spec.podNetworkInstance`
+// +kubebuilder:printcolumn:name="PodName",type=string,JSONPath=`.spec.podName`
+// +kubebuilder:printcolumn:name="UUID",type=string,JSONPath=`.status.uuid`
+// +kubebuilder:printcolumn:name="PrimaryIP",type=string,JSONPath=`.status.primaryIP`
+// +kubebuilder:printcolumn:name="MacAddress",type=string,JSONPath=`.status.macAddress`
+type MultitenantPodNetworkConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MultitenantPodNetworkConfigSpec   `json:"spec,omitempty"`
+	Status MultitenantPodNetworkConfigStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// MultitenantPodNetworkConfigList contains a list of PodNetworkConfig
+type MultitenantPodNetworkConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []MultitenantPodNetworkConfig `json:"items"`
+}
+
+// MultitenantPodNetworkConfigSpec defines the desired state of PodNetworkConfig
+type MultitenantPodNetworkConfigSpec struct {
+	// name of PNI object from requesting cx pod
+	PodNetworkInstance string `json:"podNetworkInstance,omitempty"`
+	// name of the requesting cx pod
+	PodName string `json:"podName,omitempty"`
```

Comment:
Can we call PodContext? It is podName + PodNamespace, no?

Reply:
I think we removed the pod-namespace on the component design review, because it was redundant information (the MPNC would have an ownerref to the pod, so it would always be on the same namespace)

Reply:
correct, the MTPNC namespace matches the Pod's namespace, don't need pod namespace in CRD